lib/collection/src/config.rs

```diff
@@ -161,6 +161,9 @@ pub struct CollectionConfig {
     pub wal_config: WalConfig,
     #[serde(default)]
     pub quantization_config: Option<QuantizationConfig>,
+    #[serde(default)]
+    #[schemars(skip)]
+    pub strict_mode_config: Option<StrictModeConfig>,
```

Comment:
TODO: Let's add some validation here and in `build.rs` for grpc. Specially for the fields that would be problematic to set to `0` like max limit and max timeout